Pomodoro.Api/ActionFilterAttributes/ValidateModelAttribute.cs

```diff
@@ -8,17 +8,22 @@
 namespace Pomodoro.Api.ActionFilterAttributes
 {
     /// <summary>
-    /// Action filter attribute to be used globally on API controllers for model state validation.
+    /// Action filter attribute to be used globally on API controllers for model validation.
     /// </summary>
     public class ValidateModelAttribute : ActionFilterAttribute
     {
         /// <summary>
-        /// Checks that the model state is valid.<br/>
+        /// Checks that the model is not null and its state is valid.<br/>
         /// Called before the action executes, after model binding is complete.
         /// </summary>
         /// <param name="context">The <see cref="ActionExecutingContext"/>.</param>
         public override void OnActionExecuting(ActionExecutingContext context)
         {
+            if (context.ActionArguments.Values.Any(val => val is null))
+            {
+                context.Result = new BadRequestObjectResult("The argument cannot be null.");
+            }
+
```

Comment:
How this Validation work? What's purpose of this filter?

Reply:
The next line in Program.cs, API project adds this action filter globally on controllers:

`options.Filters.Add<ValidateModelAttribute>();`

And this filter does exactly what the following code does:

`if (newcome == null || !this.ModelState.IsValid)
{
return this.BadRequest(this.ModelState);
}`

Thus, you don't have to duplicate this check from one controller method to another.

Reply:
Cool approach, but for me it's become difficult to debbug controllers. Is it possible to set a breakpoint before ModelState.IsValid check?

Reply:
Yes, you can set a breakpoint right in the `OnActionExecuting` method in the filter to check how it works.

Reply:
Thank you! My bad, couldn't do in right away)